### ISSUE TITLE
webrtc does not bind to IPv6

### ISSUE THREAD

**Issue by synnack** [AlexxIT/go2rtc#721]
I run go2rtc on an IPv6-only home assistant, and go2rtc webrtc port does not bind to IPv6 on port 8555, which it should in 2023, when the world has run out of IPv4 and I am definitely too poor for IPv4.

I've tried:
```
webrtc:
  listen: [::]:8555
```
But because it always binds on IPv4 :8555, the dual stack port is unavailable on port 8555. I *can* make it bind on 8554 or whatever other port though.

**Comment by synnack** [comment]
Turns out that 8554 is my mistake. That's the RTSP listener which works correctly for IPv6.

**Comment by synnack** [comment]
This commit broke my go2rtc:
https://github.com/AlexxIT/go2rtc/commit/e4b68518e54ff869174d08b9917153d7c36289c8 

I wonder why this commit was made..?

**Comment by synnack** [comment]
This restore commit is unfortunately incomplete: https://github.com/AlexxIT/go2rtc/commit/189f142fae25451e9e418155d462eab871e515a8

**Comment by AlexxIT** [comment]
Yeah. In 2023 it's hard to get IPv6 working in my country. Never seen it working :)

**Comment by gtxaspec** [comment]
confirmed webrtc doesn't work with ipv6.  i was wondering why I was having issues!
`tcp        0      0 0.0.0.0:8555            0.0.0.0:*               LISTEN      `

**Comment by AlexxIT** [comment]
I just need to find a way to test IPv6 live. I think go will only bind to available interfaces. Or you can use `0.0.0.0` for IPv4 or `[::]` for IPv6.

**Comment by synnack** [comment]
I don't think that currently works, [::] cannot bind if you ask for "tcp4" in webrtc.go

**Comment by gtxaspec** [comment]
@AlexxIT you can try tunnelbroker.net to get a free tunnel which will give you an ipv6 address you can use over an ipv4 connection.

**Comment by synnack** [comment]
You can also test local IPv6 by setting up radvd to use a site-local prefix (for example fc42:a1e4:417::/64 ), to prevent it from setting a default route, set AdvDefaultLifetime to a value of 0. That way you get local IPv6 in your network, without it affecting other connections. Of course the tunnelbroker solution is a better one, that way you get full IPv6 access.

**Comment by revellion** [comment]
If you need another tester for IPv6 related functionality i can assist aswell. I've got global IPv6 access from my ISP currently since a year and a half back :D

**Comment by AlexxIT** [comment]
You can test it if you want. This feature was fixed in previous releases

**Comment by imp1sh** [comment]
So is IPv6 listening possible or is it not?
